tests/unit/socket_test.cc

```diff
@@ -546,3 +547,613 @@ SEASTAR_THREAD_TEST_CASE(inet_local_remote_address_sanity) {
+            ss.abort_accept();
+        });
+        abort_timer.arm(timer<>::clock::now() + 100ms);
+
+        try {
+            auto ar = ss.accept().get();
+            got_connection = true;
+            ar.connection.shutdown_output();
+        } catch (...) {
+            // Expected - abort_accept throws
+        }
+    });
+
+    // Send header claiming 12 bytes of addresses but only send 6
+    auto client = seastar::async([&listen_addr] {
+        try {
+            auto s = connect(listen_addr).get();
+            auto out = s.output();
+            unsigned char buf[22];
+            const unsigned char sig[12] = {0x0d, 0x0a, 0x0d, 0x0a, 0x00, 0x0d, 0x0a, 0x51, 0x55, 0x49, 0x54, 0x0a};
+            std::memcpy(buf, sig, 12);
+            buf[12] = 0x21; // v2 PROXY
+            buf[13] = 0x11; // IPv4 TCP
+            buf[14] = 0x00; // length high byte
+            buf[15] = 0x0C; // length low byte (12 bytes)
+            // Only send 6 bytes of address data
+            buf[16] = 192;
+            buf[17] = 168;
+            buf[18] = 1;
+            buf[19] = 100;
+            buf[20] = 10;
+            buf[21] = 0;
+            out.write(reinterpret_cast<char*>(buf), sizeof(buf)).get();
+            out.flush().get();
+            out.close().get();
+        } catch (...) {
+            // ignore
+        }
+    });
+
+    when_all(std::move(server), std::move(client)).discard_result().get();
+    ss.abort_accept();
+
+    BOOST_REQUIRE(!got_connection);
+}
+
+SEASTAR_THREAD_TEST_CASE(proxy_protocol_v2_local_mode) {
+    listen_options lo;
+    lo.reuse_address = true;
+    lo.proxy_protocol = true;
+
+    auto listen_addr = ipv4_addr("127.0.0.1", 12004);
+    server_socket ss = seastar::listen(listen_addr, lo);
+
+    socket_address actual_remote;
+    socket_address actual_local;
+    socket_address client_actual_local;
+
+    auto server = seastar::async([&ss, &actual_remote, &actual_local] {
+        auto ar = ss.accept().get();
+        actual_remote = ar.connection.remote_address();
+        actual_local = ar.connection.local_address();
+        ar.connection.shutdown_output();
+    });
+
+    // Send LOCAL command with UNSPEC family
+    auto header = make_proxy_v2_header(true, 0x20, 0x00);  // v2 LOCAL, UNSPEC
+    auto client = seastar::async([&listen_addr, &client_actual_local, header = std::move(header)] {
+        auto s = connect(listen_addr).get();
+        client_actual_local = s.local_address();
+        auto out = s.output();
+        out.write(header.data(), header.size()).get();
+        out.flush().get();
+        out.close().get();
+    });
+
+    when_all(std::move(server), std::move(client)).discard_result().get();
+    ss.abort_accept();
+
+    // With LOCAL mode, addresses should be real connection addresses, not from header
+    // We can't check exact match since the client port is ephemeral, but we can verify
+    // that the addresses are not the fake proxy addresses and the IPs match
```

Comment:
Does the BOOST_REQUIRE_EQUAL below really implement what this comment says ("we can verify that the addresses are NOT the fake proxy addresses")? I don't see how.